asciimatics/widgets/textbox.py

```diff
@@ -114,16 +116,32 @@ def reset(self):
         # Reset to original data and move to end of the text.
         self._start_line = 0
         self._start_column = 0
-        self._line = len(self._value) - 1
+        if self._auto_scroll or self._line > len(self._value) - 1:
+            self._line = len(self._value) - 1
+
         self._column = 0 if self._is_disabled else len(self._value[self._line])
         self._reflowed_text_cache = None
 
+    def _draw_cursor(self, char, frame_no, x, y):
+        if not self._hide_cursor:
+            super(TextBox, self)._draw_cursor(char, frame_no, x, y)
+
     def _change_line(self, delta):
         """
         Move the cursor up/down the specified number of lines.
 
         :param delta: The number of lines to move (-ve is up, +ve is down).
         """
+        if self._hide_cursor:
```

Comment:
Not so keen on this bit of the change, I'm afraid.  It isn't that normal to scroll in this way just because the cursor is hidden.  Could you maybe take this bit of logic into your custom widget instead?

Reply:
Ok. My thought was the box would behave like a scrolling text box when the cursor was hidden. I'll put it in the new widget instead.